hnn_core/tests/test_viz.py

```diff
@@ -212,5 +226,123 @@ def test_dipole_visualization():
                                                   'beta_dist': 'g'})
     with pytest.raises(ValueError, match="'beta_dist' must be"):
         net.cell_response.plot_spikes_hist(color={'beta_prox': 'r'})
+    plt.close('all')
+
+
+def test_network_plotter_init(setup_net):
+    """Test init keywords of NetworkPlotter class."""
+    net = setup_net
+    # test NetworkPlotter class
+    with pytest.raises(TypeError, match='xlim must be'):
+        _ = NetworkPlotter(net, xlim='blah')
+    with pytest.raises(TypeError, match='ylim must be'):
+        _ = NetworkPlotter(net, ylim='blah')
+    with pytest.raises(TypeError, match='zlim must be'):
+        _ = NetworkPlotter(net, zlim='blah')
+    with pytest.raises(TypeError, match='elev must be'):
+        _ = NetworkPlotter(net, elev='blah')
+    with pytest.raises(TypeError, match='azim must be'):
+        _ = NetworkPlotter(net, azim='blah')
+    with pytest.raises(TypeError, match='vmin must be'):
+        _ = NetworkPlotter(net, vmin='blah')
+    with pytest.raises(TypeError, match='vmax must be'):
+        _ = NetworkPlotter(net, vmax='blah')
+    with pytest.raises(TypeError, match='trial_idx must be'):
+        _ = NetworkPlotter(net, trial_idx=1.0)
+    with pytest.raises(TypeError, match='time_idx must be'):
+        _ = NetworkPlotter(net, time_idx=1.0)
+    with pytest.raises(TypeError, match='colorbar must be'):
+        _ = NetworkPlotter(net, colorbar='blah')
```

Comment:
If you wanted a loop for these you could do it like this:

```python
not_num = 'string'
not_int = 1.0
test_kws = {'xlim': not_num, 'ylim': not_num,
            'trial_idx': not_int, 'time_idx': not_int,
            }
for keyword, value in test_kws.items():
    kwargs = {keyword: value}
    with pytest.raises(TypeError, match=f'{keyword} must be'):
        _ = NetworkPlotter(net, **kwargs)
```

But if up to you if you find it less interpretable!